src/sentry/static/sentry/app/views/organizationDiscover/resultManager.jsx

```diff
@@ -4,21 +4,52 @@
  * This function is responsible for fetching and storing result data for
  * result tables and visualizations.
  */
+import parseLinkHeader from 'app/utils/parseLinkHeader';
+
 export default function createResultManager(queryBuilder) {
-  const data = {
-    baseQuery: {query: null, data: null},
-    byDayQuery: {query: null, data: null},
-  };
+  let data = getDefault();
 
   return {
     getAll,
     fetchAll,
     reset,
     shouldDisplayResult,
+    fetchPage,
   };
 
   /**
-   * Returns data for all relevant visuzlizations.
+   * Returns data of next or previous page
+   * @param String "next" or "previous"
```

Comment:
nit: I think ` @param {String} pageToFetch` is the convention, with the type in braces and the name of the variable afterwards